container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java

```diff
@@ -180,14 +170,43 @@ private void qps(Metric.Context metricContext) {
         return context;
     }
 
+    private Metric.Context getDegradedMetricContext(String chainName, Coverage coverage) {
+        Map<DegradedReason, Metric.Context> reasons = degradedReasonContexts.get(chainName);
+        if (reasons == null) {
+            DegradedReason [] reasonEnums = {
+                    DegradedReason.MatchPhase, DegradedReason.AdaptiveTimeout,
+                    DegradedReason.Timeout, DegradedReason.NonIdealState
+            };
+            reasons = new HashMap<>(4);
+            for (DegradedReason reason : reasonEnums ) {
```

Comment:
Just use DegradedReason.values() bro ...

Reply:
That was elegant, done 👍